### PR TITLE
LNP-510: Adjudications Route + View

### PR DESCRIPTION
**Pull request by madebyzak**
This PR;
- adds the adjudications route and view
- ensures these routes/views are behind the feature flag

NB: code cleanup to follow

Encompasses:
https://dsdmoj.atlassian.net/browse/LNP-504
https://dsdmoj.atlassian.net/browse/LNP-510

### CODE REVIEW

File: server/routes/testutils/data.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { TimetableEvents } from '../../@types/launchpad'
 import { IncentiveReviewSummary } from '../../@types/incentivesApiTypes'
```
Comment (psoleckimoj) [inline server/routes/testutils/data.ts:1]: Has this actually changed? Looks like lines have just moved?

File: server/services/prisonerProfileService/index.test.ts [hunk 2]
```diff
@@ -118,6 +132,307 @@ describe('PrisonerProfileService', () => {
...
+        mockStatusQueryParam,
+      )
+    })
+  })
+
+  describe('getReportedAdjudication', () => {
+    it('should return the reported adjudication for the given charge number and agency ID', async () => {
+      const mockChargeNumber = 'ABC123'
+      const mockAgencyId = 'XYZ456'
+      const mockReportedAdjudication: ReportedAdjudicationApiResponse = {
```
Comment (psoleckimoj) [inline server/services/prisonerProfileService/index.test.ts:190]: Might make tests easier to read if test data was moved externally, especially when the objects are very large like this?

File: server/services/userService/index.test.ts [hunk 3]
```diff
@@ -14,13 +14,15 @@ describe('User service', () => {
       hmppsAuthClient = new HmppsAuthClient(null) as jest.Mocked<HmppsAuthClient>
       userService = new UserService(hmppsAuthClient)
     })
+
```
Comment (psoleckimoj) [inline server/services/userService/index.test.ts:17]: No actual changes? Is it a linter doing this?

File: server/views/components/adjudications/hearing/offences/punishments/template.njk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  <tbody class="govuk-table__body">
+    {% for punishment in punishments %}
+      <tr class="govuk-table__row">
+        <td class="govuk-table__cell">{{punishment.privilegeType}}</td>
+        <td class="govuk-table__cell sanction_no_wrap">{{punishment.schedule.days}} days</td>
+        <td class="govuk-table__cell sanction_no_wrap">{{punishment.effectiveDate}}</td>
+      </tr>
+    {% endfor %}
+  </tbody>
+</table>
```
Comment (psoleckimoj) [inline server/views/components/adjudications/hearing/offences/punishments/template.njk:23]: Missing trailing newline

File: server/views/components/adjudications/hearing/offences/template.njk [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        value: {
+          text: hearing.outcome.details
+        }
+      }
+    ]
+  }) }}
+
+  {% if hearing.punishments | length %}
+    {{ adjudicationHearingOffencePunishments(hearing.punishments) }}
+  {% endif %}
```
Comment (psoleckimoj) [inline server/views/components/adjudications/hearing/offences/template.njk:48]: Missing trailing newline

File: server/views/components/adjudications/hearing/template.njk [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+              text: hearing.outcome.details
+            }
+          }
+        ]
+      }) }}
+
+      {{ adjudicationHearingOffences(hearing) }}
+    </section>
+  </article>
+</div>
```
Comment (psoleckimoj) [inline server/views/components/adjudications/hearing/template.njk:50]: Missing trailing newline

File: server/views/pages/profile.njk [hunk 7]
```diff
@@ -71,14 +74,18 @@
         {% endif %}
       </div>
     </div>
-
+
```
Comment (psoleckimoj) [inline server/views/pages/profile.njk:77]: Linter would normally have picked this up 🤔

**Review by psoleckimoj — APPROVED** [review]
LGTM!

Only comments are try and break the PRs up into smaller chunks so it's easier to read and less daunting. Like put all your constants, config, types etc that don't actually affect anything in one PR. Then you could do the data layer next we nothing is depending on that, then the service layer, then the routes etc.

File: server/routes/testutils/data.ts (hunk 1 above)
Comment (madebyzak) [reply server/routes/testutils/data.ts:1]: Yeh I think this is just an import re-order.

File: server/services/prisonerProfileService/index.test.ts (hunk 2 above)
Comment (madebyzak) [reply server/services/prisonerProfileService/index.test.ts:190]: Agreed - that's the plan with the cleanup PR, as I've done exactly that with the transactions and visits code. This one got quite messy and was already quite a big PR when I picked it up from Barry so it definitely needs re-aligning with the other code I've written, but didn't want to make this one any bigger!

File: server/services/userService/index.test.ts (hunk 3 above)
Comment (madebyzak) [reply server/services/userService/index.test.ts:17]: I think it's just a line-break between the `it()` blocks - not sure whether it's something I've done manually for readability or whether it's the linter.

File: server/views/pages/profile.njk (hunk 7 above)
Comment (madebyzak) [reply server/views/pages/profile.njk:77]: Yeh, true 🤔

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**